chia/consensus/block_body_validation.py

```diff
@@ -180,36 +180,25 @@ async def validate_block_body(
 
             # The generator_refs_root must be the hash of the concatenation of the List[uint32]
             generator_refs_hash = std_hash(
-                b"".join([(i).to_bytes(4, byteorder="big") for i in block.transactions_generator_ref_list])
+                b"".join([i.to_bytes(4, byteorder="big") for i in block.transactions_generator_ref_list])
             )
             if block.transactions_info.generator_refs_root != generator_refs_hash:
                 return Err.INVALID_TRANSACTIONS_GENERATOR_REFS_ROOT, None
             if len(block.transactions_generator_ref_list) > constants.MAX_GENERATOR_REF_LIST_SIZE:
                 return Err.PRE_SOFT_FORK_TOO_MANY_GENERATOR_REFS, None
 
         if block.transactions_generator is not None:
-            # The generator must be less than MAX_GENERATOR_SIZE bytes in length
-            if len(bytes(block.transactions_generator)) > constants.MAX_GENERATOR_SIZE:
-                return Err.PRE_SOFT_FORK_MAX_GENERATOR_SIZE, None
-
-            # Get List of names removed, puzzles hashes for removed coins and conditions created
-            if cached_cost_result is not None:
-                result: Optional[CostResult] = cached_cost_result
-            else:
-                result = calculate_cost_of_program(block.transactions_generator, constants.COST_PER_BYTE)
-            # The call to calculate cost runs the generator program
-            if result is None:
-                return Err.INVALID_COST_RESULT, None
+            # Get List of names removed, puzzles hashes for removed coins and conditions crated
 
-            assert result is not None
-            cost = result.cost
-            npc_list = result.npc_list
+            assert npc_result is not None
+            cost = calculate_cost_of_program(block.transactions_generator, npc_result, constants.COST_PER_BYTE)
+            npc_list = npc_result.npc_list
 
             # 8. Check that cost <= MAX_BLOCK_COST_CLVM
             if cost > constants.MAX_BLOCK_COST_CLVM:
                 return Err.BLOCK_COST_EXCEEDS_MAX, None
-            if result.error is not None:
-                return Err(result.error), None
+            if npc_result.error is not None:
+                return Err.GENERATOR_RUNTIME_ERROR, None
```

Comment:
Could be good to have more granularity here... why not just return `npc_result.error`?